chart/scripts/post-delete-hook.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -e
+
+succeeded=()
+failed=()
+echo "Uninstalling Rancher resources in the following namespaces:" "$namespaces"
+for namespace in $namespaces; do
+  app_list=$(helm list -n "$namespace" -q)
+  for app in $app_list; do
+    if [[ $app =~ .crd$ ]]; then
+      echo "--- Skip the app [$app] in the namespace [$namespace]"
+      continue
+    fi
+    echo "--- Deleting the app [$app] in the namespace [$namespace]"
+    if [[ ! $(helm uninstall "$app" -n "$namespace") ]]; then
+      failed=("${failed[@]}" "$app")
+      continue
+    fi
+    t=0
+    # by default, wait 120 seconds in total for an app to be uninstalled
+    while [[ "$(kubectl get pods --selector app="$app" -n "$namespace" -o json | jq '.items | length')" != "0" ]] && [[ $t -lt "$timeout" ]]; do
+      echo "waiting 5 seconds for pods of [$app] to be terminated ..."
+      sleep 5
+      t=$((t + 5))
+    done
+    if [[ "$(kubectl get pods --selector app="$app" -n "$namespace" -o json | jq '.items | length')" != "0" ]]; then
+      echo "timeout uninstalling [$app] in the namespace [$namespace]"
+      failed=("${failed[@]}" "$app")
+    else
+      echo "successfully uninstalled [$app] in the namespace [$namespace]"
+      succeeded=("${succeeded[@]}" "$app")
+    fi
+  done
```

Comment:
Can we drop this in favor of just adding the `--timeout` flag on the helm uninstall?

https://helm.sh/docs/helm/helm_uninstall

Reply:
We'd have to trap the exit code so we can list all failures. Or do away with that idea.

Reply:
Talked with Jack offline and we are going to not use timeout as it doesn't buy us anything but extending default timeout of 5 min.